### PR TITLE
feat: add wrapper types for all integer/float (de)serialization

### PR DESCRIPTION
**Pull request by wyfo**

### CODE REVIEW

**Comment by wyfo** [comment]
All `Int8`/`UInt8`/etc. has been added.
The question is now: should we keep support for `int` and `float`? In this case, it must be clearly documented than `int` would be equivalent to `Int64` (arbitrary choice), and `float` to `Float64` (arbitrary choice).
Otherwise, the use can add the support with only 4 lines of Python:
```python
serializer(lambda i: ZBytes.serialize(Int64(i)), target=int)
deserializer(lambda zbytes: zbytes.deserialize(Int64), target=int)
serializer(lambda f: ZBytes.serialize(Float64(f)), target=float)
deserializer(lambda zbytes: zbytes.deserialize(Float64), target=float)
```

**Comment by Mallets** [comment]
I'd go for keeping `int` and `float` and mapping it to `Int64` and `Float64`, which is kind of the same behaviour of `usize` on `64bit` architecture on Rust.
It needs to be clearly documented.

**Comment by wyfo** [comment]
I've added `ZBytes` documentation. `z_bytes.py` example is in the PR following.

File: zenoh/__init__.py [hunk 1]
```diff
@@ -12,3 +12,89 @@
...
+_INT64_MIN = -(1 << 63)
+_INT128_MIN = -(1 << 127)
+
+_INT8_MAX = 1 << 7
+_INT16_MAX = 1 << 15
+_INT32_MAX = 1 << 31
+_INT64_MAX = 1 << 63
+_INT128_MAX = 1 << 127
+
+_UINT8_MAX = 1 << 8
```
Comment (Mallets) [inline zenoh/__init__.py:28]: As the naming stands, these values are not correct. `_UINTX_MAX` should be actually `(1 << X) - 1`.

Comment (wyfo) [reply zenoh/__init__.py:28]: Same for `_INTX_MAX` btw, but I use `<` for max, so it's ok, as they are private constants.

**Review by Mallets — APPROVED** [review]

File: zenoh/__init__.pyi [hunk 2]
```diff
@@ -54,7 +90,7 @@ class Config:
     def from_file(cls, path: str | Path) -> Self: ...
     @classmethod
     def from_json5(cls, obj: Any) -> Self: ...
-    @property
+    property
```
Comment (Mallets) [inline zenoh/__init__.pyi:93]: Is this a wanted change or some simple error?

Comment (wyfo) [reply zenoh/__init__.pyi:93]: It's a typo indeed, I'm fixing it

Comment (wyfo) [inline zenoh/__init__.pyi:93]:
```suggestion
    @property
```

**Review by Mallets — APPROVED** [review]

**Pull request merged**